package/src/hooks/useSyncClientEvents.ts

```diff
@@ -0,0 +1,69 @@
+import { useCallback } from 'react';
```

Comment:
Since we are now using quite a few of these, perhaps it makes sense to create a builder hook that is going to generate stuff automatically.

For example, instead of `channel` we could have `data` or something and make it as generic as possible (so that if we want to use it for other stuff we freely can).

Reply:
I will take this up later and try to refine it to a single hook. Currently, it is different from the previous one for membership state as here we listen to client events and there we did listen to channel events. But this should be doable.